sacrebleu/tokenizers/tokenizer_13a.py

```diff
@@ -1,17 +1,17 @@
-# -*- coding: utf-8 -*-
-
-from .tokenizer_none import NoneTokenizer
+from functools import lru_cache
+from .tokenizer_base import BaseTokenizer
 from .tokenizer_re import TokenizerRegexp
 
 
-class Tokenizer13a(NoneTokenizer):
+class Tokenizer13a(BaseTokenizer):
 
     def signature(self):
         return '13a'
 
     def __init__(self):
         self._post_tokenizer = TokenizerRegexp()
 
+    @lru_cache(maxsize=None)
```

Comment:
What is the use case (exact command) when lru_cache should help?

My benchmarks (wmt17 de-en) give me slightly faster times (and less memory) when excluding this lru_cache
here and also in tokenizer_re and tokenizer_intl.
(My expectation was that the cache in tokenizer_re is extra, if it is called only from another function which is already cached.)
Interestingly, this holds also for `--paired bs` and `-i newstest2017/de-en/*` with 11 systems, so there is a chance that some sentences will be the same across different systems and the cache will be used.

Reply:
My mind is confused on this as well. I initially added this as a quick hack so that references are not re-tokenized over and over when using multi-system mode. But then I realized that this wasn't the good way to do it and I switched to reference caching. But weirdly, if I remove the caches, I slightly have slower runtimes. I thought that the reason was what you wrote i.e. some hypotheses are the same across systems. But you're also saying that removing the caches are faster for you? Sorry, didn't understand your case, could you clarify?

Also, how do you measure the time? I used `scripts/perf_test.py` once on the branch and once after manually commenting all lru_cache's.

Reply:
Interesting: `scripts/perf_test.py` uses `newstest2017/cs-en`, which caching makes a bit faster. I used `newstest2017/de-en`, which caching makes a bit slower.
I tried python `timeit` and also extrinsic (running it several times and computing mean/median):
`/usr/bin/time -f "\t%E real,\t%M mem" sacrebleu/sacrebleu.py -t wmt17 -l de-en -i data/wmt17-submitted-data/txt/system-outputs/newstest2017/de-en/*`.

The differences are not big, but caching could help with more system outputs (or test sets with many repeated sentences).  The memory is also not much bigger, so we can keep the cache there.